Pod/Classes/NSAttributedString+BonMotUtilities.m

```diff
@@ -0,0 +1,88 @@
+
+@implementation NSAttributedString (BonMotUtilities)
+
+- (NSString *)humanReadableString
+{
+    NSString *originalString = self.string;
+    NSMutableString *composedHumanReadableString = [NSMutableString string];
+
+    [originalString enumerateSubstringsInRange:NSMakeRange(0, originalString.length) options:NSStringEnumerationByComposedCharacterSequences usingBlock:^(NSString *BONCNullable substring, NSRange substringRange, NSRange enclosingRange, BOOL *BONCNonnull stop) {
+
+        static NSCharacterSet *s_newLineCharacterSet = nil;
+        if (!s_newLineCharacterSet) {
+            s_newLineCharacterSet = [NSCharacterSet newlineCharacterSet];
+        }
+        NSCharacterSet *s_whiteSpaceAndNewLinesSet = nil;
+        if (!s_whiteSpaceAndNewLinesSet) {
+            s_whiteSpaceAndNewLinesSet = [NSCharacterSet whitespaceAndNewlineCharacterSet];
+        }
+
+        // Substitute Images with @"{image#heightx#width}"
+        if ([substring isEqualToString:BONSpecial.objectReplacementCharacter]) {
+            BONStringDict *attributes = [self attributesAtIndex:substringRange.location effectiveRange:NULL];
+            NSTextAttachment *attachment = attributes[NSAttachmentAttributeName];
+            UIImage *attachedImage = attachment.image;
+            NSString *imageSubstitutionString = [NSString stringWithFormat:@"{image%.0fx%.0f}", attachedImage.size.height, attachedImage.size.width];
+            [composedHumanReadableString appendString:imageSubstitutionString];
+        }
+        else {
+            // Substitute BONSpecial Characters with @"{#name}"
+            unichar character = [substring characterAtIndex:0];
+            BONGeneric(NSDictionary, NSNumber *, NSString *)*specialNames = @{
+                @(BONCharacterLineFeed) : @"{lineFeed}",
+                @(BONCharacterTab) : @"{tab}",
+                @(BONCharacterSpace) : @" ",
+                @(BONCharacterNoBreakSpace) : @"{noBreakSpace}",
+                @(BONCharacterEnSpace) : @"{enSpace}",
+                @(BONCharacterEmSpace) : @"{emSpace}",
+                @(BONCharacterFigureSpace) : @"{figureSpace}",
+                @(BONCharacterThinSpace) : @"{thinSpace}",
+                @(BONCharacterHairSpace) : @"{hairSpace}",
+                @(BONCharacterZeroWidthSpace) : @"{zeroWidthSpace}",
+                @(BONCharacterNonBreakingHyphen) : @"{nonBreakingHyphen}",
+                @(BONCharacterFigureDash) : @"{figureDash}",
+                @(BONCharacterEnDash) : @"{enDash}",
+                @(BONCharacterEmDash) : @"{emDash}",
+                @(BONCharacterHorizontalEllipsis) : @"{horizontalEllipsis}",
+                @(BONCharacterLineSeparator) : @"{lineSeparator}",
+                @(BONCharacterParagraphSeparator) : @"{paragraphSeparator}",
+                @(BONCharacterNarrowNoBreakSpace) : @"{narrowNoBreakSpace}",
+                @(BONCharacterWordJoiner) : @"{wordJoiner}",
+                @(BONCharacterMinusSign) : @"{minusSign}"
```

Comment:
#pp trailing comma here, so that if another item is added later, it doesn't dirty this line in Git history.